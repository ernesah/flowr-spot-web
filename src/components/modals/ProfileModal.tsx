import { useContext } from 'react';
import Button from '../Button';
import { AuthContext } from '../../store/auth-context';
import avatar from '../../assets/images/profile-holder.png';

const UserInfo = ({ label, value }: { label: string; value: string }) => {
  return (
    <div className='mb-3'>
      <label className='text-dusty-grey text-xs mb-4'>{label}</label>
      <h5 className='text-lg'>{value}</h5>
    </div>
  );
};

const ProfileModal: React.FC<{
  closeModal: () => void;
}> = ({ closeModal }) => {
  const { user, logoutUser } = useContext(AuthContext);

  return (
    <div className='flex flex-col items-center gap-y-4 mb-5'>
      <div className='flex items-center gap-x-6'>
        <img src={avatar} alt='avatar' className='w-20 h-20' />
        <div>
          <h3 className='text-2xl font-light'>
            {user?.first_name} {user?.last_name}
          </h3>
          <p className='font-light text-sm text-dusty-grey'>47 sightings</p>
        </div>
      </div>
      <div className='items-start my-4'>
        <UserInfo label='First Name' value={user?.first_name || ''} />
        <UserInfo label='Last Name' value={user?.last_name || ''} />
        <UserInfo label='Date of Birth' value='May 20, 1980' />
        <UserInfo label='Email Address' value='michael.berry@gmail' />
      </div>
      <Button
        title='Logout'
        variant='primary'
        classes='rounded px-8'
        handleClick={() => {
          closeModal();
          logoutUser();
        }}
      />
    </div>
  );
};

export default ProfileModal;
