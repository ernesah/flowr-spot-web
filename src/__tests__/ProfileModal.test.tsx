import { render } from '@testing-library/react';
import ProfileModal from '../components/modals/ProfileModal';

describe('ProfileModal component', () => {
  test('matches snapshot', () => {
    const closeModal = jest.fn();
    const { asFragment } = render(<ProfileModal closeModal={closeModal} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
