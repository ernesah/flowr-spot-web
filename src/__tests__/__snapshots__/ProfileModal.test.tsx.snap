// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileModal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center gap-y-4 mb-5"
  >
    <div
      class="flex items-center gap-x-6"
    >
      <img
        alt="avatar"
        class="w-20 h-20"
        src="profile-holder.png"
      />
      <div>
        <h3
          class="text-2xl font-light"
        >
           
        </h3>
        <p
          class="font-light text-sm text-dusty-grey"
        >
          47 sightings
        </p>
      </div>
    </div>
    <div
      class="items-start my-4"
    >
      <div
        class="mb-3"
      >
        <label
          class="text-dusty-grey text-xs mb-4"
        >
          First Name
        </label>
        <h5
          class="text-lg"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="text-dusty-grey text-xs mb-4"
        >
          Last Name
        </label>
        <h5
          class="text-lg"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="text-dusty-grey text-xs mb-4"
        >
          Date of Birth
        </label>
        <h5
          class="text-lg"
        >
          May 20, 1980
        </h5>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="text-dusty-grey text-xs mb-4"
        >
          Email Address
        </label>
        <h5
          class="text-lg"
        >
          michael.berry@gmail
        </h5>
      </div>
    </div>
    <button
      class="text-sm font-medium bg-soft-pink-gradient text-white shadow-soft-pink py-3 px-5 rounded px-8"
      type="button"
    >
      Logout
    </button>
  </div>
</DocumentFragment>
`;
